File: api-dashboard-frontend/src/services/apiService.js
import axios from 'axios';

const BASE_URL = "http://localhost:8080/api";

export const getCatFact = async () => {
  const response = await axios.get(`${BASE_URL}/catfact`);
  return response.data;
};

export const getJoke = async () => {
  const response = await axios.get(`${BASE_URL}/joke`);
  return response.data;
};

export const getRandomUser = async (count = 1) => {
  const response = await axios.get(`${BASE_URL}/randomuser?count=${count}`);
  return response.data;
};

File: apiDashboard_Backend/src/main/java/com/example/apiDashboard/service/ApiService.java
package com.example.apiDashboard.service;

import com.example.apiDashboard.model.CatFact;
import com.example.apiDashboard.model.Joke;
import com.example.apiDashboard.model.RandomUser;
import org.springframework.stereotype.Service;
import org.springframework.web.client.RestTemplate;

@Service
public class ApiService {

    private final RestTemplate restTemplate = new RestTemplate();

    public CatFact getCatFact() {
        String url = "https://catfact.ninja/fact";
        return restTemplate.getForObject(url, CatFact.class);
    }

    public Joke getJoke() {
        String url = "https://official-joke-api.appspot.com/random_joke";
        return restTemplate.getForObject(url, Joke.class);
    }

    public RandomUser getRandomUser() {
        String url = "https://randomuser.me/api/";
        return restTemplate.getForObject(url, RandomUser.class);
    }
}
